Mathlib/NumberTheory/ModularForms/SlashInvariantForms.lean

```diff
@@ -3,29 +3,71 @@ Copyright (c) 2022 Chris Birkbeck. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Chris Birkbeck
 -/
-import Mathlib.NumberTheory.ModularForms.CongruenceSubgroups
+import Mathlib.NumberTheory.ModularForms.ArithmeticSubgroups
 import Mathlib.NumberTheory.ModularForms.SlashActions
 
 /-!
 # Slash invariant forms
 
 This file defines functions that are invariant under a `SlashAction` which forms the basis for
 defining `ModularForm` and `CuspForm`. We prove several instances for such spaces, in particular
-that they form a module.
+that they form a module over `ℝ`, and over `ℂ` if the group is contained in `SL(2, ℝ)`.
 -/
 
+open Complex UpperHalfPlane ModularForm
 
-open Complex UpperHalfPlane
-
-open scoped MatrixGroups ModularForm
+open scoped MatrixGroups
 
 noncomputable section
 
+section
+
+variable {R n : Type*} [CommRing R] [Fintype n] [DecidableEq n] (Γ : Subgroup (GL n R))
+
+/-- Typeclass saying that a subgroup of `GL(2, ℝ)` has determinant contained in `{±1}`. Necessary
+so that the typeclass system can detect when the slash action is multiplicative. -/
+class HasDetPlusMinusOne : Prop where
```

Comment:
i feel this `HasDet..` class would maybe fit better somewhere else, could it go in the ArithmeticSubgroups file?

Reply:
Good point, will move them.